Independer.WCFDataAnnotations/ValidatingParameterInspector.cs

```diff
@@ -85,8 +89,32 @@ public class ValidatingParameterInspector : IParameterInspector {
       return null;
     }
 
-    private IEnumerable<IObjectValidator> GetValidators(int parameterPosition) {
-      return _parameterDetailsInfo != null && _parameterDetailsInfo.ParameterDetails.Single(x => x.Position == parameterPosition).SkipNullcheck
+    private IEnumerable<ValidationResult> AddParameterNamesWhenMissing(IEnumerable<ValidationResult> validationResults, string parameterName) {
+      if (validationResults == null || !validationResults.Any() || String.IsNullOrWhiteSpace(parameterName)) {
+        return validationResults;
+      }
+
+      var extendedValidationResults = new List<ValidationResult>();
+
+      foreach (var validationResult in validationResults) {
+        extendedValidationResults.Add(validationResult.MemberNames == null || !validationResult.MemberNames.Any() 
+          ? new ValidationResult(validationResult.ErrorMessage, new List<string> { parameterName }) 
+          : validationResult);

```

Comment:
when we do have some parameterNames sets, don't we also want to ensure we got the current one in there too?
like if !validationResults.MemberNames.Contains(parameterName) then add?

Reply:
Makes sense, I will add it